git-subhistory.sh

```diff
@@ -1,6 +1,11 @@
-#!/bin/sh
+#!/bin/bash
 # http://github.com/laughinghan/git-subhistory
 
+# Request bash to report that a pipeline failed if any of the commands
+# in the pipe failed (not just the last one).  This is necessary for
+# proper error handling.
+set -o pipefail
+
```

Comment:
```
* Force using bash and enable the 'pipefail' option to catch failures of
  git filter-branch that were being masked by piping through say_stdin
  (which always succeeded).
```

ooh, good catch. However, I don't want to force bash (if I could, I would've just done `set -o errexit`), for this to be considered for `git/contrib` it needs to follow Git's shell guidelines: https://github.com/git/git/blob/master/Documentation/CodingGuidelines

what if we replace all instances of `say_stdin` with:
```
output=$(git filter-branch ... 2>&1) || exit $?
say "$output"
```